### PR TITLE
RDM-2714: tech debt - migrate state override in callback.

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
RDM-2714: tech debt - migrate state override in callback.
   [RDM-2714] (https://tools.hmcts.net/jira/browse/RDM-2714).

### CODE REVIEW

File: wiremock/mappings/no_state_about_to_submit.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "request": {
+    "method": "POST",
+    "urlPath": "/case_type/no_state/about_to_submit"
```
Comment (mario-paniccia) [inline wiremock/mappings/no_state_about_to_submit.json]: we should not have case type names and callback names on the path. To allow these json to be fully reusable in other case types and callbacks.
Also try to reuse as much as possible json files already existing rather than introducing new ones.
You could rename this urlPath to:
`callback_returning_only_data`

File: wiremock/mappings/state_inside_about_to_submit.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "request": {
+    "method": "POST",
+    "urlPath": "/case_type/state/inside/about_to_submit"
```
Comment (mario-paniccia) [inline wiremock/mappings/state_inside_about_to_submit.json]: rename to 
`callback_overriding_state_as_data`

File: wiremock/mappings/top_state_about_to_submit.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "request": {
+    "method": "POST",
+    "urlPath": "/case_type/top/state/about_to_submit"
```
Comment (mario-paniccia) [inline wiremock/mappings/top_state_about_to_submit.json]: rename to
```
callback_overriding_state_as_top_level_property

File: wiremock/mappings/top_state_no_data_about_to_submit.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+  "request": {
+    "method": "POST",
+    "urlPath": "/case_type/top/state/no_data/about_to_submit"
+  },
+  "response": {
+    "status": 200,
+    "headers": {
```
Comment (mario-paniccia) [inline wiremock/mappings/top_state_no_data_about_to_submit.json]: do we need to introduce this file? Can't we test this corner scenario with a unit test rather than an integration test?

File: wiremock/mappings/two_state_about_to_submit.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+  "request": {
+    "method": "POST",
+    "urlPath": "/case_type/two/state/about_to_submit"
```
Comment (mario-paniccia) [inline wiremock/mappings/two_state_about_to_submit.json]: you should be able to reuse one of the previous files you already introduced, since the json response is identical

File: wiremock/mappings/top_state_no_data_about_to_submit.json (hunk 4 above)
Comment (Thor-tech-of-metal) [reply wiremock/mappings/top_state_no_data_about_to_submit.json]: We have already test it in a the integration test . But I think that it would be nice having this scenario in the stubs for further use and to complete the whole test coverage.

Comment (Thor-tech-of-metal) [reply wiremock/mappings/top_state_no_data_about_to_submit.json]: it is already cover in the unit test.  Nonetheless I think that it would be nice having this test to cover all scenarios in the integration test.

File: wiremock/mappings/two_state_about_to_submit.json (hunk 5 above)
Comment (Thor-tech-of-metal) [reply wiremock/mappings/two_state_about_to_submit.json]: This is a  quite different case which it would useful adding it. there is not a previous existing mock that emulate this response with 2 states.    I think that it would be nice having this test to cover all scenarios in Case Multiple Pages the integration test.
